src/posit/connect/_deprecated.py

```diff
@@ -0,0 +1,23 @@
+import warnings
+
+
+# From https://github.com/posit-dev/py-shiny/blob/f6b92d8cf49a90f3b3dbb636cd6d7fdeee244cfd/shiny/_deprecated.py#L15C1-L28C1
+# Create our own warning class instead of using built-in DeprecationWarning, because we
+# want to be able to control display of these messages without interfering with the
+# user's control of DeprecationWarning.
+# 2024-11: Change base class to DeprecationWarning
```

Comment:
Is this a behavior we want to override? 

There does appear to be some churn in how silent these are: https://docs.python.org/3/library/warnings.html#default-warning-filter but it smells a bit funny to me to wrap around core language functionality / expectations. Not saying we _absolutely shouldn't_ but this might be a larger conversation than this PR?

Reply:
Issue created #344 for followup work